overlord/snapstate/snapmgr.go

```diff
@@ -530,7 +533,31 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 }
 
 func (m *SnapManager) undoPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
-	// FIXME: remove the entire function
+	st := t.State()
+	st.Lock()
+	snapsup, err := TaskSnapSetup(t)
+	st.Unlock()
+	if err != nil {
+		return err
+	}
+
+	// report this error to an error tracker
+	if osutil.GetenvBool("SNAPPY_TESTING") {
+		return nil
+	}
+	if snapsup.SideInfo.RealName == "" {
+		return nil
+	}
+	st.Lock()
+	var logMsg []string
+	for _, t := range t.Change().Tasks() {
+		if t.Status() == state.ErrorStatus {
+			logMsg = append(logMsg, t.Log()...)
+		}
+	}
+	st.Unlock()
+	errtracker.Report(snapsup.SideInfo.RealName, snapsup.SideInfo.Channel, strings.Join(logMsg, "\n"))
```

Comment:
Suggestion:
 - collect this report and store it in the state
 - have a errormgr that has a task that on an exponential backoff sends unsent reports to daisy
 - the manager can be also used to purge old reports eventually
 - we could use a new internal command that would dump them to stdout in some useful form